src/coreComponents/constitutive/fluid/multifluid/compositional/functions/RachfordRice.hpp

```diff
@@ -151,12 +140,11 @@ struct RachfordRice
     // step 3: start the Newton loop
     integer newtonIteration = 0;
     real64 newtonValue = gasPhaseMoleFraction;
+    real64 funcNewton = evaluate( kValues, feed, presentComponentIds, newtonValue );
 
     while( ( currentError > newtonTolerance ) && ( newtonIteration < maxNewtonIterations ) )
     {
-      real64 const deltaNewton = -evaluate( kValues, feed, presentComponentIds, newtonValue )
-                                 / evaluateDerivative( kValues, feed, presentComponentIds, newtonValue );
-      currentError = LvArray::math::abs( deltaNewton ) / LvArray::math::abs( newtonValue );
+      real64 const deltaNewton = -funcNewton / evaluateDerivative( kValues, feed, presentComponentIds, newtonValue );
 
       // test if we are stepping out of the [xMin;xMax] interval
       if( newtonValue + deltaNewton < xMin )
```

Comment:
```suggestion
      if( newtonValue + deltaNewton < xMin )
      {
        deltaNewton = 0.5 * ( xMin - newtonValue );
      }
      else if( newtonValue + deltaNewton > xMax )
      {
        deltaNewton = 0.5 * ( xMax - NewtonValue );
      }
       
      newtonValue = newtonValue + deltaNewton;
```

This way, when the `deltaNewton` is checked in `currentError` below, it is accurately reflecting the `deltaNewton` used in case of chopping at the bounds.